File: 2020/day1/part2.go
package main

import (
    "bufio"
    "fmt"
    "log"
	"os"
	"strconv"
)

func main() {

	file, err := os.Open("input.txt")
    if err != nil {
        log.Fatal(err)
    }
	defer file.Close()
	
	var expenseReport []int

    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
		// fmt.Println(scanner.Text())
		inp := scanner.Text()
		intInp,conErr := strconv.Atoi(inp)
		if conErr != nil {
			log.Fatal(conErr) 
		}
		expenseReport = append(expenseReport,intInp)
	}
	// fmt.Println(expenseReport)

	for index, first := range expenseReport {
		for secondIndex, second := range expenseReport[index:] {
			for _, third := range expenseReport[secondIndex:] {
				if first + second + third == 2020 {
					fmt.Println(first*second*third)
					return
				}
			}
		}
	}


    if err := scanner.Err(); err != nil {
        log.Fatal(err)
    }

}

File: 2020/day1/part2.py
# expenseReport = [1721,
#                  979,
#                  366,
#                  299,
#                  675,
#                  1456]

# cache = {}
def main():
    expenseReport = []
    inputFile = open("input.txt", "r")
    for line in inputFile:
        expenseReport.append(int(line))

    for index, first in enumerate(expenseReport):
        for secondIndex, second in enumerate(expenseReport[index:]):
            for third in expenseReport[secondIndex:]:
                if first + second + third == 2020:
                    print(first * second * third)
                    return

        # cache[first] = second
main()
